RestaurantManagementSystem/RestaurantManagementSystem/src/UseCases/Email Confirmation/EmailConfirmationHandler.cs

```diff
@@ -6,33 +6,28 @@ namespace RestaurantManagementSystem.src.UseCases.Email_Confirmation
 {
     public class EmailConfirmationHandler : IEmailConfirmationHandler
     {
-        private readonly IRestaurantRepo _restaurantRepo;
+        private readonly IRegistrationRequestRepo _restaurantRepo;
         private readonly IEmailService _emailservice;
-        public EmailConfirmationHandler (IRestaurantRepo restaurantRepo,IEmailService emailservice)
+        public EmailConfirmationHandler (IRegistrationRequestRepo restaurantRepo,IEmailService emailservice)
         {
             _restaurantRepo = restaurantRepo;
             _emailservice = emailservice;
         }
-        public async Task<bool> Confirm(ConfirmationCommand confirmationCommand)
+        public async Task<bool> VerifyEmail(ConfirmationCommand confirmationCommand)
         {
-            Guid id = new Guid(confirmationCommand.RequestId);
-            RegistrationRequest obj = await _restaurantRepo.FindByIdAsync(id);
-            if (obj.Status == RequestStatus.Status.PendingVerification)
-            {
-                RegistrationRequest Updated = RegistrationRequest.VertifyCode(confirmationCommand.ConfirmationCode, obj);
-                if (Updated.Status == RequestStatus.Status.WaitingForAdminResponse)
+            RegistrationRequest obj = await _restaurantRepo.FindByIdAsync(confirmationCommand.RequestId);
```

Comment:
what is obj don't every name variable like this? this is a `registrationRequest` and also it should not be called restaurantRepo it's registartionRequestRepo, but you can refactor later